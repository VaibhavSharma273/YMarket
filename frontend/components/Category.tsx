import React from 'react';
import { View, Image, Text, StyleSheet, TouchableOpacity } from 'react-native';
import { text_styles } from './TextNormalize';


const Category = ({ category, navigation }: any) => {
    // onPress={() => navigation.push('SearchCategory', { category: category.title})}
    return (
        <View style={styles.listItemShadow}>
            <TouchableOpacity style={styles.listItem}>
                <View style={styles.listItemFooter}>
                    <Text style={[ text_styles.large, { fontWeight: "600" } ]} 
                          onPress={() => navigation.push('SearchCategory', { category: category.title})}>
                        {category.title}
                    </Text>
                </View>
            </TouchableOpacity>
        </View>
    )
}

const styles = StyleSheet.create({
    listItem: {
      borderRadius: 16,
      backgroundColor: "#fff",
      marginTop: 10,
      marginBottom: 10,
      marginLeft: 20,
      marginRight: 20,
    },
    listItemShadow: {
      borderRadius: 16,
      backgroundColor: 'transparent',
      shadowColor: '#000',
      shadowOffset: {
        width: 0,
        height: 1,
      },
      shadowOpacity: 0.22,
      shadowRadius: 2.22,
      elevation: 3,
     },
    listItemFooter: {
        flexDirection: "row",
        padding: 8,
        paddingLeft: 16,
    },
  });

export default Category