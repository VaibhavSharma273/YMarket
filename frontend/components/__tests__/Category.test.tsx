import * as React from 'react';
import renderer from 'react-test-renderer';

import Category from '../Category';

it(`renders correctly`, () => {
    const props = {
        category: jest.fn(),
        navigation: jest.fn(),
    };
    const tree = renderer.create(<Category {...props} />).toJSON();

    expect(tree).toMatchSnapshot();
});
