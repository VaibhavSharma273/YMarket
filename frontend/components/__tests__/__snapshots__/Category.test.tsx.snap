// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "transparent",
      "borderRadius": 16,
      "elevation": 3,
      "shadowColor": "#000",
      "shadowOffset": Object {
        "height": 1,
        "width": 0,
      },
      "shadowOpacity": 0.22,
      "shadowRadius": 2.22,
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#fff",
        "borderRadius": 16,
        "marginBottom": 10,
        "marginLeft": 20,
        "marginRight": 20,
        "marginTop": 10,
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
          "padding": 8,
          "paddingLeft": 16,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "fontSize": 43,
            },
            Object {
              "fontWeight": "600",
            },
          ]
        }
      />
    </View>
  </View>
</View>
`;
